neatle/src/main/java/si/inova/neatle/operation/OperationImpl.java

```diff
@@ -174,10 +174,9 @@ private void executeNext() {
             }
 
             Command old = currentCommand;
-            currentCommand = commandQueue.poll();
+            currentCommand = commandQueue.isEmpty() ? EMPTY_COMMAND : commandQueue.poll();
```

Comment:
Now that the `currentCommand` is properly synchronized you don't need to change this (check for empty).